In [24]:
# Importing required libraries
import pandas as pd

In [25]:
# Loading our attrition data into a pandas dataframe
df = pd.read_csv(
    "../data/raw/WA_Fn-UseC_-HR-Employee-Attrition.csv"
)

# Visualizing top 5 records
df.head()

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2


In [26]:
# Checking the shape of the dataset
print(f"The dataset contains {df.shape[0]} rows and {df.shape[1]} columns")

The dataset contains 1470 rows and 35 columns


In [27]:
# Checking data type and unique values from all columns
df_summary = pd.DataFrame({
    'Data Type': df.dtypes,
    'Unique': df.nunique()
})
df_summary

,Data Type,Unique
Age,int64,43
Attrition,str,2
BusinessTravel,str,3
DailyRate,int64,886
Department,str,3
DistanceFromHome,int64,29
Education,int64,5
EducationField,str,6
EmployeeCount,int64,1
EmployeeNumber,int64,1470


In [28]:
# Checking some stats from all numerical columns 
df.describe().T # .T tansposes the output showing the collumns as rows


,count,mean,std,min,25%,50%,75%,max
Age,1470.0,36.923810,9.135373,18.0,30.00,36.0,43.00,60.0
DailyRate,1470.0,802.485714,403.509100,102.0,465.00,802.0,1157.00,1499.0
DistanceFromHome,1470.0,9.192517,8.106864,1.0,2.00,7.0,14.00,29.0
Education,1470.0,2.912925,1.024165,1.0,2.00,3.0,4.00,5.0
EmployeeCount,1470.0,1.000000,0.000000,1.0,1.00,1.0,1.00,1.0
EmployeeNumber,1470.0,1024.865306,602.024335,1.0,491.25,1020.5,1555.75,2068.0
EnvironmentSatisfaction,1470.0,2.721769,1.093082,1.0,2.00,3.0,4.00,4.0
HourlyRate,1470.0,65.891156,20.329428,30.0,48.00,66.0,83.75,100.0
JobInvolvement,1470.0,2.729932,0.711561,1.0,2.00,3.0,3.00,4.0
JobLevel,1470.0,2.063946,1.106940,1.0,1.00,2.0,3.00,5.0


In [29]:
# Filtering just the columns with 'str' datatypes
df_summary[df_summary['Data Type'] == 'str']

,Data Type,Unique
Attrition,str,2
BusinessTravel,str,3
Department,str,3
EducationField,str,6
Gender,str,2
JobRole,str,9
MaritalStatus,str,3
Over18,str,1
OverTime,str,2


#### **Conclusion**:
- The exploratory analysis shows that the dataset contains **1,470** employee records across **35** variables, providing a broad view of employee demographics, compensation, job characteristics, satisfaction, and career experience. **The numerical variables reveal substantial variation** in factors such as **age**, **income**, **tenure**, **distance from home**, and **years of experience**. Satisfaction and workplace-related measures generally fall around the middle of their respective scales, while **categorical variables provide useful segmentation** by **department**, **job role**, **business travel**, **education field**, and **marital status**. Some constant variables, such as EmployeeCount, Over18, and StandardHours, provide little analytical value and may be excluded. Overall, the dataset provides a strong foundation for investigating employee characteristics, workplace factors, and potential drivers of attrition.